lib/control-connection.js

```diff
@@ -95,10 +95,12 @@ ControlConnection.prototype.init = function (callback) {
           var portSeparatorIndex = name.lastIndexOf(']:');
           return addHost(name.substr(1, portSeparatorIndex - 1), name.substr(portSeparatorIndex + 2), eachNext);
         }
-        if (name.indexOf('.') > 0 && name.indexOf(':') > 0) {
+        if (name.indexOf(':') > 0) {
```

Comment:
With the previous code `resolveAll` would be called with `my-host:9999` as `name` which wouldn't resolve anything from a DNS lookup.

What I was thinking was that if we only passed in the host part to resolveAll and then provided the port with each resolved IP address this would work and this way we would consistently always use DNS to resolve host names to IPs (whether we provide the port or not).   Here's [a patch](https://gist.github.com/tolbertam/6f27ca431ade44107b30915eccd47a94#file-nodejs-355-resolvewithport-patch) of what I was thinking.